src/Orleans/Utils/Pool/PooledResource.cs

```diff
@@ -54,7 +32,7 @@ public virtual void SignalPurge()
         /// </summary>
         public void Dispose()
         {
-            IObjectPool<T> localPool = Interlocked.Exchange(ref pool, null);
```

Comment:
To bring another perspective, the type may not matter, but having an explicit way to cue in a casual read (especially when viewing in GH without VS) is helpful to form a mental picture &ndash; to decrease cognitive pressure.